crates/c/src/lib.rs

```diff
@@ -1061,10 +1052,13 @@ void __wasm_export_{ns}_{snake}_dtor({ns}_{snake}_t* arg) {{
         }
     }
 
-    fn type_alias(&mut self, id: TypeId, _name: &str, ty: &Type, docs: &Docs) {
+    fn type_alias(&mut self, id: TypeId, name: &str, ty: &Type, docs: &Docs) {
+        // we should skip generating `typedef` for `Resource` types because they aren't even
+        // defined anywhere, not even in `type_resource`. Only its `Handle` types are defined.
+        // The aliasing handle types are defined in `define_anonymous_type`.
         let target = dealias(self.resolve, id);
         if !matches!(&self.resolve.types[target].kind,
-                     TypeDefKind::Resource if self.gen.resources[&target].direction == Direction::Import)
+            TypeDefKind::Resource if self.gen.resources[&target].direction == Direction::Import)
         {
```

Comment:
I was poking at this as well and I agree that this is needed for now with the comment you had above. While you're here though could you change this to drop the `if .. == Direction::Import`? This is equally applicable to exports as well as imports.

Additionally it might be a bit more clear to invert this by doing:

```rust
if matches!(..., Resource) {
   return;
}

self.src.h_defs(...);
// ...
```